src/reducers/ThemeSlice.ts

```diff
@@ -0,0 +1,20 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { ThemeMode } from '../types/state';
+
+const initialState = {
+  mode: localStorage.getItem('theme') === 'dark' ? 'dark' : 'light',
```

Comment:
nit: add `as ThemeMode` at the end, to force `mode` to use the type `ThemeMode` (currently it looks like it's infered as a generic string)
Alternatively you can type initialState explicitely:
```js
type State = {
  mode: ThemeMode
};
const initialState: State = {
  mode: ...
};
```